app/src/main/java/com/jnj/vaccinetracker/visit/screens/VisitDosingFragment.kt

```diff
@@ -52,50 +57,66 @@ class VisitDosingFragment : BaseFragment(),
         binding = DataBindingUtil.inflate(inflater, R.layout.fragment_visit_dosing, container, false)
         binding.viewModel = viewModel
         binding.lifecycleOwner = viewLifecycleOwner
-        binding.root.setOnClickListener { activity?.currentFocus?.hideKeyboard() }
 
+        setupFilters()
+        setupClickListeners()
+        setupInputListeners()
+
+        return binding.root
+    }
+
+    private fun setupFilters() {
+        binding.editTextWeightInput.filters = arrayOf(InputFilterMinMax(MIN_WEIGHT, MAX_WEIGHT))
+    }
+
+    private fun setupClickListeners() {
+        binding.root.setOnClickListener { activity?.currentFocus?.hideKeyboard() }
         binding.imageButtonBarcodeScanner.setOnClickListener {
-            startActivityForResult(ScanBarcodeActivity.create(requireContext(),ScanBarcodeActivity.MANUFACTURER), REQ_SCAN_BARCODE)
+            startActivityForResult(ScanBarcodeActivity.create(requireContext(), ScanBarcodeActivity.MANUFACTURER), REQ_SCAN_BARCODE)
         }
         binding.btnSubmit.setOnClickListener {
             submitDosingVisit()
         }
         binding.dropdownManufacturer.setOnItemClickListener { _, _, position, _ ->
-            val manufacturerName = viewModel.manufacturerList.get()?.distinct()?.get(position) ?: return@setOnItemClickListener
+            val manufacturerName = viewModel.manufacturerList.get()?.distinct()?.get(position)
+                    ?: return@setOnItemClickListener
             viewModel.setSelectedManufacturer(manufacturerName)
         }
+    }
+
+    private fun setupInputListeners() {
         binding.editVialBarcode.doOnTextChanged { s, _, _, _ ->
             if (!s.isNullOrEmpty()) {
                 viewModel.matchBarcodeManufacturer(s, resourcesWrapper)
-            }
-             else{
+            } else {
                 viewModel.setSelectedManufacturer("")
                 binding.dropdownManufacturer.clearListSelection()
-                }
+            }
+        }
```

Comment:
This looks good already, What are the conditions for the app to show obese.
![Screenshot 2024-06-10 074102](https://github.com/johnsonandjohnson/vxnaid/assets/48015736/4d1173c0-d027-4890-b5c1-76a629d5c9c7)

Reply:
The exact formula is not know yet. It shows random values for now.